src/group_theory/solvable.lean

```diff
@@ -175,4 +227,39 @@ begin
   ... = ⊥ : rfl,
 end
 
+lemma is_solvable_of_top_eq_bot (h : (⊤ : subgroup G) = ⊥) : is_solvable G :=
+⟨⟨0, by simp *⟩⟩
+
+variables {G} {G' : Type*} [group G'] {f : G →* G'}
+
+lemma solvable_of_solvable_injective (hf : function.injective f) (h : is_solvable G') :
+  is_solvable G :=
+begin
+  rw is_solvable_def at *,
+  cases h with n hn,
+  use n,
+  rw eq_bot_iff_map_eq_bot hf,
+  rw eq_bot_iff at *,
+  calc map f (derived_series G n) ≤ derived_series G' n : map_derived_series_le_derived_series f n
+  ... ≤ ⊥ : hn,
+end
+
+lemma subgroup_solvable_of_solvable (H : subgroup G) (h : is_solvable G) : is_solvable H :=
+solvable_of_solvable_injective (show function.injective (subtype H), by ext) h
```

Comment:
I think you can use the lemma `subtype.val_injective`

Reply:
Yeah, that works. I can't get rid of the `show function.injective (subtype H)` though because Lean can't figure out what `subtype.val_injective` should be applied to.